server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -73,6 +73,17 @@ public IngestDocument(String index, String type, String id, String routing, Stri
         this.ingestMetadata.put(TIMESTAMP, ZonedDateTime.now(ZoneOffset.UTC));
     }
 
+    public IngestDocument(String index, String type, String id, String routing,String parent,
```

Comment:
Instead of adding a new constructor, can the existing constructor be modified to also accept version and versionType? I know this requires changing more tests, but I think this class should only have two constructors.